cli/auth.go

```diff
@@ -83,6 +83,11 @@ type tokenResponse struct {
 	TokenType    string `json:"token_type"`
 }
 
+type authMethod interface {
+	fmt.Stringer
+	dialOpts() rpc.DialOption
+}
+
```

Comment:
Can you add compile time assertions here that both `token` and `apiKey` implement `authMethod`?

Reply:
:+1: . Please LMK if there is a better way to do compile time assertions.